### PR TITLE
Change shoc 1d views to have type Scalar

### PR DESCRIPTION
**Pull request by tcclevenger**
A bit of code clean up. 1d views in shoc were type `view<ekat::Pack<Scalar,1>>`, now they are type `view<Scalar>`. Just simplifies the code a bit.

### CODE REVIEW

File: components/scream/src/physics/shoc/shoc_functions.hpp [hunk 1]
```diff
@@ -74,9 +73,9 @@ struct Functions
     SHOCInput() = default;
 
     // Grid spacing of host model in x direction [m]
-    view_1d<const Pack1d> host_dx;
+    view_1d<const Scalar> host_dx;
```
Comment (jeff-cohere) [inline components/scream/src/physics/shoc/shoc_functions.hpp:76]: This might be a silly question, but why do we sometimes call this data type a `Scalar` and other times a `Real`?

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 2]
```diff
@@ -1799,7 +1826,7 @@ void check_length_scale_shoc_length_f(Int nlev, Int shcol, Real* host_dx, Real*
   // Sync to device
   std::vector<view_1d> temp_1d_d(2);
   std::vector<view_2d> temp_2d_d(1);
-  ekat::host_to_device({host_dx,host_dy}, shcol, temp_1d_d);
+  copy_to_device({host_dx,host_dy}, shcol, temp_1d_d);
   ekat::host_to_device({shoc_mix}, shcol, nlev, temp_2d_d, true);
```
Comment (jeff-cohere) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:1830]: Any reason not to use `copy_to_device` here too?

**Review by jeff-cohere — APPROVED** [review]
Looks good to me.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp (hunk 2 above)
Comment (tcclevenger) [reply components/scream/src/physics/shoc/shoc_functions_f90.cpp:1830]: `copy_to_device` is a function I wrote for 1d scalar views. `shoc_mix` is a packed view, so it still requires the ekat function.

File: components/scream/src/physics/shoc/shoc_functions.hpp (hunk 1 above)
Comment (tcclevenger) [reply components/scream/src/physics/shoc/shoc_functions.hpp:76]: Good question.. I'm not quite sure why the shoc `Functions` structure allows for a different scalar type that Real in scream_types..

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 3]
```diff
@@ -843,6 +843,42 @@ void pblintd(PblintdData& d)
 
 // end _c impls
 
+// Helper function to copy 1d data to and from Device
+template<typename ViewT>
+void copy_to_device(const std::vector<typename ViewT::value_type const*>& data,
```
Comment (bartgol) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:848]: So, the only big difference between this fcn and what's in ekat (beside the fact that you have a fixed size for all views, which is a minor difference) is that your views are `view<Real*>` rather than `view<Pack<T,N>*>`, right?

Would it make sense to put these utilities in `scream/src/share/util`? Something like `scream_deep_copy.hpp`. I suspect other parts of scream might need this at some point.

Also (but not for this PR, I think), we might have a simpler version:
```
template<typename T>
void deep_copy(const view_1d<T>& dst, typename ViewT::value_type const* src);
template<typename T>
void deep_copy(typename ViewT::value_type const* dst, const view_1d<T>& src);

template<typename T>
void deep_copy(const view_2d<T>& dst, typename ViewT::value_type const* src, const int src_stride);
template<typename T>
void deep_copy(typename ViewT::value_type const* dst, const view_2d<T>& src, const int dst_stride);
```
and have the 'vector' version simply call this. We should also use enalbe_if to provide an overload if T=Pack<S,N>, cause in that case we need the 'physical' length, to avoid reading/writing out of bounds.
The single view version can be maybe a tad easier to visually parse, when syncing 1 or 2 views:
```
view_1d a(ncols);
view_2d b(ncols,nlevs), c(ncols,nlevs);
deep_copy(a, a_data);
deep_copy(b, b_data, nlevs);
deep_copy(c, c_data, nlevs);
```
I feel like creating the vector kind of adds a bit of clunkiness, and doesn't save many lines of code for a handful of views.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 4]
```diff
@@ -1047,7 +1080,7 @@ void update_host_dse_f(Int shcol, Int nlev, Real* thlm, Real* shoc_ql, Real* exn
   std::vector<const Real*> ptr_array = {thlm,  shoc_ql, exner, zt_grid, host_dse};
 
   // Sync to device
-  ekat::host_to_device({phis}, shcol, temp_1d_d);
+  copy_to_device({phis}, shcol, temp_1d_d);
```
Comment (bartgol) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:1083]: Follow up on the comment above: here's an example where having to create a vector makes the code a bit less obvious, imho. It'd be easier to parse if we could do
```
view_1d phis_d(shcol);
copy_to_device(phis_d,phis);
```

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 5]
```diff
@@ -1934,25 +1959,25 @@ void shoc_diag_obklen_f(Int shcol, Real* uw_sfc, Real* vw_sfc, Real* wthl_sfc, R
...
+  using Spack   = typename SHOC::Spack;
   using Scalar  = typename SHOC::Scalar;
-  using view_2d = typename SHOC::view_2d<Pack1>;
-  using view_1d = typename SHOC::view_1d<Pack1>;
+  using view_2d = typename SHOC::view_2d<Spack>;
+  using view_1d = typename SHOC::view_1d<Scalar>;
 
-  std::vector<Int> dim1(2, shcol);
-  std::vector<Int> dim2  = {nlevi,  nlev};
+  std::vector<Int> dim1 = {shcol, shcol};
```
Comment (bartgol) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:1979]: This change seems to be unrelated to this PR. Was it a bug you caught?

**Review by bartgol — APPROVED** [review]
Looks good. I have some comments, but nothing major.

**Comment by jgfouca** [comment]
@tcclevenger , just curious, why are shoc views not using SIMD Packs?

**Comment by bartgol** [comment]
> @tcclevenger , just curious, why are shoc views not using SIMD Packs?

The ones he changed are 2d fields on the surface (which means 1d view of size ncols). For those, packing does not buy you anything. And if by chance we turned one of those into a strided subview of a 3d field (i.e., 2d view), it might even be wrong.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp (hunk 5 above)
Comment (tcclevenger) [reply components/scream/src/physics/shoc/shoc_functions_f90.cpp:1979]: Yeah, I didn't mean to include that. I deleted that line (evidentially) and wrote it back slightly different. I'll change back.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp (hunk 3 above)
Comment (tcclevenger) [reply components/scream/src/physics/shoc/shoc_functions_f90.cpp:848]: Yes, I think putting this in `scream/src/share/util` makes sense. I'll move it there.

**Comment by tcclevenger** [comment]
@bartgol I moved the copy _to* functions to scream/../utils. In a later PR I'll do the impl of the simpler version

**Comment by jgfouca** [comment]
@bartgol , gotcha. So these views have 1 scalar per column.

**Review by jgfouca — APPROVED** [review]
